field_utils/FieldCondensedArray_private.F90

```diff
@@ -26,9 +26,9 @@ function get_fptr_shape_private(gridToFieldMap, localElementCount, has_vertical,
       
       vert_dim = 0
       vert_size = 1
-      
+
       rank = size(localElementCount)
-      grid_dims = pack(gridToFieldMap, gridToFieldMap /= 0)
+      grid_dims = pack(gridToFieldMap, gridToFieldMap > 0)
```

Comment:
Can you elaborate on the circumstances where elements of gridToFieldMap are negative?   My memory is that the documentation says "0" or the index of ...

Reply:
For the field below,  `gridToFieldMap` is [ 1, 2, -1]
```
Field Print Starts ====>
 Field base status = Ready
   Name = 'Field034'
 Base name    = Field034
 Status: Base = Ready,  object = Ready
 Proxy        = no
 Root Info (Attributes):
{
  "MAPL": {
    "internal": {
      "long_name": "unknown",
      "standard_name": "E",
      "ungridded_dims": {
        "num_ungridded_dimensions": 0
      },
      "units": "m",
      "vertical_dim": {
        "vloc": "VERTICAL_DIM_CENTER"
      },
      "vertical_grid": {
        "num_levels": 3
      }
    }
  }
} Field status =            4
 gridrank =            2
--- ESMCI::Array::print start ---
 Base name    = Field034
 Status: Base = Ready,  object = Ready
 Proxy        = no
 Root Info (Attributes):
{}Array typekind/rank: Real*4 / 3
~ lower class' values ~
DistGrid dimCount = 2
deCount = 1
localDeCount = 1
~ local data in LocalArray for DE 0 ~
LocalArrayPrint: Array at address 0x23fe6f0:
            rank = 3, typekind = 5, base_addr = 0x14aab7580100
            dim[0] = 12  dim[1] = 13  dim[2] = 3
  Real, *4, Dim 3, Data values:
( 1, 1, 1) =  17
  skipping to end ...
(12,13, 3) =  17
dim 1: [0]: [1 [1 [1, 12] 12] 12]
dim 2: [1]: [1 [1 [1, 13] 13] 13]
dim 3: undistLBound[0]=1            undistUBound[0]=3
--- ESMCI::Array::print end ---
 arrayrank =            3
 gridToFieldMap ungriddedLBound ungriddedUBound totalLWidth totalUWidth
           1           1           3                0           0
           2          -1          -1                0           0
          -1          -1          -1               -1          -1
          -1          -1          -1               -1          -1
          -1          -1          -1               -1          -1
          -1          -1          -1               -1          -1
          -1          -1          -1               -1          -1
 Field Print Ends   ====>
```

Reply:
`gridToFieldMap` should only have 2 elements in for this field:

List with number of elements equal to the grid’s dimCount.

This suggests the bug is elsewhere when allocating the array.

Reply:
I think the problem is in the higher level routine.   `dimCount` was used to allocate the array but it should be `geomDimCount`.    Not that the man page makes the distinction particularly clear.

Reply:
@darianboggs - please determine why the existing tests did not cover this case (and then help @pchakraborty to remedy the tests as part of this PR.)

Reply:
> I think the problem is in the higher level routine. `dimCount` was used to allocate the array but it should be `geomDimCount`. Not that the man page makes the distinction particularly clear.

I have pushed this change.